app/models/user.rb

```diff
@@ -47,6 +46,12 @@ def send_activation_email
     UserMailer.account_activation(self).deliver_now
   end
 
+  def password_valid?(value)
```

Comment:
This should be rewritten to be a custom validation instead of just a method in the User model. http://guides.rubyonrails.org/active_record_validations.html#custom-methods

Look for other examples under app/validators